src/ar_http_iface_h.erl

```diff
@@ -17,13 +18,28 @@ init(InitialReq, State) ->
 %%% Private functions.
 %%%===================================================================
 
-do_handle(Req) ->
+%% @doc Because cowboy 2.0 seemingly doesn't support accessing the request from
+%% multiple processes (see https://github.com/ninenines/cowboy/issues/1374) the
+%% following is the only way to introduce a asynchronous handler timeout.
+handler_timeout_after(Req, T) ->
+	HandlerPid = self(),
+	spawn_link(fun() ->
+		timer:sleep(T),
+		ar:warn([
+			handler_timeout,
+			{method, cowboy_req:method(Req)},
+			{path, cowboy_req:path(Req)}
+		]),
+		exit(HandlerPid, kill)
```

Comment:
`kill` is not a good reason. Something with "timeout" would be more meaningful.

Reply:
`kill` is actually the system atom required to kill the process which traps exits.